includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1247,7 +1247,12 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return string
 	 */
 	public function generate_upe_checkout_experience_accepted_payments_html( $key, $data ) {
-		$stripe_account      = $this->stripe_request( 'account' );
+		try {
+			$stripe_account = $this->stripe_request( 'account' );
+		} catch ( WC_Stripe_Exception $e ) {
+			WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
```

Comment:
Shouldn't we bail out, although gracefully, if $stripe_account triggers an exception?

This variable is used below. I'd expect some errors down this method if an exception is thrown by $this->stripe_request( 'account' ) because $stripe_account won't be the object this method expects.

Reply:
As it is gracefully handled below with `isset( $stripe_account->capabilities )` and `isset( $stripe_capabilities[ $capability_id ] )` checked, we should not get any error even if `$stripe_account` got an exception. We are good here :sunglasses:

Reply:
Right! Scratch that. Thanks for clarifying it!

Btw, do you see who invokes this method? I was trying to find the call stack but I'm missing who calls it.

Reply:
Looks like it's getting rendered by WooCommerce. WooCommerce [generates the HTML](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-settings-api.php#L348) for settings based on [the type](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-settings-api.php#L345). On the Stripe end the type is being [set here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/fix/displaying-of-prb-wallets/includes/admin/stripe-settings.php#L285).